data/hlint.yaml

```diff
@@ -113,10 +113,9 @@
     - warn: {lhs: last (sortBy f x), rhs: maximumBy f x, side: isCompare f}
     - warn: {lhs: reverse (sortBy f x), rhs: sortBy (flip f) x, name: Avoid reverse, side: isCompare f, note: Stabilizes sort order}
     - warn: {lhs: sortBy (flip (comparing f)), rhs: sortOn (Down . f)}
-    - warn: {lhs: sortBy (comparing f), rhs: sortOn f, side: notEq f fst && notEq f snd}
+    - warn: {lhs: sortBy (comparing f), rhs: sortOn f, side: notEq f fst && notEq f snd && notEq f Down && notEq f Data.Ord.Down, note: Only appropriate if f is expensive enough to justify the cost of caching its results. See the documentation for sortOn}
```

Comment:
I think rather than warning about a fairly complex condition, it might be better to just delete this hint entirely

Reply:
I agree. While there are obviously cases where `sortOn` is the right choice, a user should probably be specifically wanting to optimize their expensive sort and hopefully come across `sortOn` in their own research. Covered in 86363703feaa7e1abf2d41fe458512f3943beab4.